torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -2468,6 +2468,9 @@ def _get_param_to_unflat_param_names(
     parameters, these mapped-to lists always contain a single element. The
     unflattened parameter names should match the keys of the model state dict.
 
+    For shared parameters, only the first parameter name is included (following
```

Comment:
This follows the `nn.Module.parameters()` convention and suffices for the current use cases of this function.

Reply:
Do we know why this didn't come up in unittests? I recall we have a module that uses shared params in transformer module in `common_fsdp.py`. Can we add a unittest to catch this in the future?

Reply:
In my unit tests, I did exercise code paths that call this helper method on the `TransformerWithSharedParams` *after* wrapping with `FSDP`. The issue arises when calling this method on a non-wrapped `nn.Module` with shared parameters. I only noticed the `assert` failing for the `TransformerWithSharedParams` when using this helper in the `FSDP` constructor *before* wrapping.

I will see how to best incorporate this case into the unit tests. It was certainly an oversight on my end.